### PR TITLE
First draft: A Better Way To Consult

### PR DESCRIPTION
**Pull request by moggyboy**

### CODE REVIEW

File: _posts/2016-09-06-a-better-way-to-consult.markdown [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+layout: trikeapps_show
+title:  "A better way to consult"
+date:   2016-09-06 11:38:00 +1000
+categories: matt_fallshaw
+---
+## Time vs Features, and standard contract forms
+
+The thing that software developers do is convert time spent working on software features into
+delivered software features. Software developers' clients generally care about exactly one of those
+things, but due to complications discussed in this essay they often end up paying for the other one.
```
Comment (jack-trikeapps) [inline _posts/2016-09-06-a-better-way-to-consult.markdown]: Omit needless words: strike "due to complications discussed in this essay". It's implied from the essay structure and described explicitly in the next sentence.

File: _posts/2016-09-06-a-better-way-to-consult.markdown [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+Let's look at those two contracts.
+
+### The Time and Materials contract
+
+The fundamental problem with the Time and Materials contract is obvious to anyone who finds
+themselves contemplating becoming a client to a software developer – you agree to pay the developer
+enough for their time that they're making a profit on each hour, and the longer it takes the
+developer to deliver the features you want, the more profit they make. Your developer now has a
+direct financial incentive to work slowly- (I know – it's more complicated than that and there are
```
Comment (jack-trikeapps) [inline _posts/2016-09-06-a-better-way-to-consult.markdown]: Move the parenthetical into a new sentence and probably remove some of these hyphens/wannabe-emdashes.

File: _posts/2016-09-06-a-better-way-to-consult.markdown [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+  larger the job looks, and this isn't the most efficient way to go about discovering that extra
+  detail!
+* Even if you take the time to describe what you want, seeing it often convinces you that you want
+  something different - we're just not very good at imagining what we'll like. Our brains, highly
+  tuned by evolution to gather berries and hunt mammoth, just aren't adept at foretelling the
+  software features we'll actually like when we see them.
+* Seeing your first few ideas implemented can will give you new ideas - you start the project with
+  an idea of how everything will fit together... but the prototype teaches you a lot! If you're
+  building and developing a prototype this happens over and over again. In addition, there are lots
+  of things you can do to help during the project... and you should have an incentive to help
```
Comment (jack-trikeapps) [inline _posts/2016-09-06-a-better-way-to-consult.markdown]: Missing `.`

File: _posts/2016-09-06-a-better-way-to-consult.markdown [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+So, you start with a Fixed Cost contract and a fixed view of what you want, and you either stick
+with that (to the detriment of your project you ignore all of the good ideas you come up with
+during the project), or you start discussing the variations. Here the incentives are pretty awful
+too – the contractor offers a price for each variant, and the client pretty much has to accept it
+(because you're already deep in the project).
+
+Fixed Cost contracts thus often end up either:
+
+* delivering what was asked for at the beginning of the project instead of the much better design
```
Comment (jack-trikeapps) [inline _posts/2016-09-06-a-better-way-to-consult.markdown]: Are you starting your bullet points with capitals or not?

File: _posts/2016-09-06-a-better-way-to-consult.markdown [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+* Describing what you want is hard, and takes a lot of time - the more detail you go into as you
+  describe something, the more detail you see... the longer you spend trying to define the job the
+  larger the job looks, and this isn't the most efficient way to go about discovering that extra
+  detail!
+* Even if you take the time to describe what you want, seeing it often convinces you that you want
+  something different - we're just not very good at imagining what we'll like. Our brains, highly
+  tuned by evolution to gather berries and hunt mammoth, just aren't adept at foretelling the
+  software features we'll actually like when we see them.
+* Seeing your first few ideas implemented can will give you new ideas - you start the project with
```
Comment (jack-trikeapps) [inline _posts/2016-09-06-a-better-way-to-consult.markdown]: A lot of this rehashes the previous point. I'd refocus it towards the benefits of rapid prototyping or something like that.

File: _posts/2016-09-06-a-better-way-to-consult.markdown [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+  building and developing a prototype this happens over and over again. In addition, there are lots
+  of things you can do to help during the project... and you should have an incentive to help
+* You're concerned that our incentives are bad in a Time and Materials contract - you having an
+  incentive to help usually makes a big difference to how quickly we can work. We need you for
+  clarification of requirements, conversation about options, your acceptance of our work when we
+  ship it, and your helpful criticism when you don't want to accept our work. All of this requires
+  your time, and it's highly likely you have other competing demands on your time.
+
+So, you start with a Fixed Cost contract and a fixed view of what you want, and you either stick
+with that (to the detriment of your project you ignore all of the good ideas you come up with
```
Comment (jack-trikeapps) [inline _posts/2016-09-06-a-better-way-to-consult.markdown]: reword to reduce run-on: "you ignore all the good ideas you come up with, to the project's detriment"

File: _posts/2016-09-06-a-better-way-to-consult.markdown [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+Fixed Cost contracts thus often end up either:
+
+* delivering what was asked for at the beginning of the project instead of the much better design
+  that could have grown out of what everyone learned during the project; or
+* wasting time and effort in contract renegotiations
+
+### Trike's contract form, and why it's better
+
+If we cut the two classic contract forms exactly in half and add in a splash of the
+[Lean](http://en.wikipedia.org/wiki/Lean_software_development) and
```
Comment (jack-trikeapps) [inline _posts/2016-09-06-a-better-way-to-consult.markdown]: Unsure whether these should be Proper Nouns.

File: _posts/2016-09-06-a-better-way-to-consult.markdown [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+hourly rate, then call half of that number the feature completion bonus (paid only once you, the
+client, are happy that the feature actually fulfils your requirements), and charge half our hourly
+rate for the hours we actually work.
+
+## An example
+
+Assume for the sake of easy round numbers that we charge $100 per hour (we don't).
+
+We estimate a feature at 10 hours of work, and therefore estimate it at $1000 and quote it at
+$500 + $50/hr. You accept the quote because you recognise that the feature delivers you much more
```
Comment (jack-trikeapps) [inline _posts/2016-09-06-a-better-way-to-consult.markdown]: s/recognise/believe/?

File: _posts/2016-09-06-a-better-way-to-consult.markdown [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+In a Time and Materials contract you've paid for some of the developer's time. If more time is
+required to make the feature work then it's reasonable that you pay for that.
+**Cost to fix the bug: hours worked × full contractors hourly rate (so the contractor is earning as
+  much for fixing the bug as they did for the original work).**
+
+If you discover a bug in a feature you paid for on a Shared Cost basis... well, fixing the bug is
+part of the time required to get the feature to “finished”, right? And you're paying for that
+time... so it's reasonable that you pay for fixing the bug, but only at the reduced hourly rate
+you're paying for the developer's time.
+**Cost to fix the bug: hours worked × half the contractor's nominal hourly rate (so the contractor
```
Comment (jack-trikeapps) [inline _posts/2016-09-06-a-better-way-to-consult.markdown]: I would set the equations separately from the explanations of the equations' implications.

File: _posts/2016-09-06-a-better-way-to-consult.markdown (hunk 1 above)
Comment (moggyboy) [inline _posts/2016-09-06-a-better-way-to-consult.markdown]: Fixed.

File: _posts/2016-09-06-a-better-way-to-consult.markdown (hunk 2 above)
Comment (moggyboy) [inline _posts/2016-09-06-a-better-way-to-consult.markdown]: Fixed.

File: _posts/2016-09-06-a-better-way-to-consult.markdown (hunk 3 above)
Comment (moggyboy) [inline _posts/2016-09-06-a-better-way-to-consult.markdown]: Added `.`

File: _posts/2016-09-06-a-better-way-to-consult.markdown (hunk 4 above)
Comment (moggyboy) [inline _posts/2016-09-06-a-better-way-to-consult.markdown]: Fixed

File: _posts/2016-09-06-a-better-way-to-consult.markdown (hunk 5 above)
Comment (moggyboy) [inline _posts/2016-09-06-a-better-way-to-consult.markdown]: Fixed.

**Pull request merged**